lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js

```diff
@@ -35,86 +31,7 @@ const FULL_COVERAGE = 1;
-    const missingVerificationDisplay = missingVerificationsCount
-        ? h(
-            '.d-inline-block.va-t-bottom',
-            tooltip(h('.f7', iconWarning()), missingVerificationMessage),
-        )
-        : null;
-
-    const notBadPercentageFormat = h('span', formatFloat((1 - badEffectiveRunCoverage) * 100, { precision: 0 }));
-
-    const nonMcReproducibleQcSummaryContent = [
-        notBadPercentageFormat,
-        missingVerificationDisplay,
-    ];
-
-    /**
-     * Format QC summary with color
-     * @param {*} content content
-     * @param {string} [configuration.color] the color to look for contrast,
-     * in hexa RGB or CSS var() function over variable name expression: e.g. "var(--color-nice)"
-     * @returns {Component} QC summary display
-     */
-    const getQcSummaryDisplayWithColor = (content, { color }) => h(
-        '.btn.w-100',
-        { style: {
-            backgroundColor: color,
-            color: isContrastBlack(/var\(.+\)/.test(color) ? getValueFromCssVar(color.slice(4, -1)) : color) ? 'black' : 'white',
-        } },
-        content,
-    );
-
-    let qcSummaryDisplay = null;
-    if ([badEffectiveRunCoverage, explicitlyNotBadEffectiveRunCoverage].includes(null)) {
-        qcSummaryDisplay = getQcSummaryDisplayWithColor(
-            [
-                h('.d-inline-block', tooltip(
-                    iconBolt(),
-                    'Run start or stop is missing and time-based flag was assigned, coverage cannot be calculated',
-                )),
-                missingVerificationDisplay,
-            ],
-            {
-                color: QC_SUMMARY_COLORS.INCALCULABLE_COVERAGE,
-            },
-        );
-    } else if (!badEffectiveRunCoverage) {
-        qcSummaryDisplay = getQcSummaryDisplayWithColor(
-            nonMcReproducibleQcSummaryContent,
-            {
-                color: explicitlyNotBadEffectiveRunCoverage === FULL_COVERAGE
-                    ? QC_SUMMARY_COLORS.ALL_GOOD
-                    : QC_SUMMARY_COLORS.PARTIALLY_GOOD_NOT_BAD,
-            },
-        );
-    } else if (!explicitlyNotBadEffectiveRunCoverage && mcReproducible) {
-        qcSummaryDisplay = getQcSummaryDisplayWithColor(
-            [
-                notBadPercentageFormat,
-                h('em.d-inline-block.va-top.f7', 'MC.R'),
-                missingVerificationDisplay,
-            ],
-            { color: QC_SUMMARY_COLORS.LIMITED_ACCEPTANCE_MC_REPRODUCIBLE },
-        );
-    } else {
-        qcSummaryDisplay = getQcSummaryDisplayWithColor(
-            nonMcReproducibleQcSummaryContent,
-            {
-                color: badEffectiveRunCoverage === FULL_COVERAGE
-                    ? QC_SUMMARY_COLORS.ALL_BAD
-                    : QC_SUMMARY_COLORS.PARTIALLY_BAD,
-            },
-        );
-    }
+    let qcSummaryDisplay = getQcSummaryDisplay(qcSummary[runNumber][dplDetectorId]);
```

Comment:
I think it is best to not declare and generate the content of the summary display as a let here and then reassign based on a ternary condition.
Instead, only call the `getQcSummaryDisplay` in the ternary condition if it meets the criteria (L38)

Reply:
I would say, it will decrease readability

Reply:
Why do you think it will descrease readability? You are already applying the `tooltip` in the ternary already rather than having it as a `const`. 

Would you say that this is better 
```js
let qcSummaryDisplay = getQcSummaryDisplay(qcSummary[runNumber][dplDetectorId]);
 qcSummaryDisplay = isRunNotSubjectToQc(run)
        ? tooltip(qcSummaryDisplay, getRunNotSubjectToQcReason(run))
        : qcSummaryDisplay;
```
than this?
```js
const qcSummaryDisplay = isRunNotSubjectToQc(run)
        ? tooltip(qcSummaryDisplay, getRunNotSubjectToQcReason(run))
        : getQcSummaryDisplay(qcSummary[runNumber][dplDetectorId]);
```

To me is the exact opposite for readability + the benefit of efficiency